packages/dd-trace/src/platform/node/loader.js

```diff
@@ -57,7 +57,13 @@ class Loader {
 
         pkg = require(`${basedir}/package.json`)
 
-        const mainFile = path.posix.normalize(pkg.main || 'index.js')
+        let mainFile = path.posix.normalize(pkg.main || 'index.js')
+
+        // Support libs that specify a folder as their `main` module, e.g.: `{ "main": "./lib") }`.
+        if (!path.posix.extname(mainFile)) {
```

Comment:
Node also supports files without extension, so I think the logic should be similar to that of `require` itself.

Reply:
Pushed again. Went with `require.resolve()` so as to not reimplement the require logic (which is actually a lot more complex than I had imagined 😅 )